gnucash/gnome/window-reconcile.c

```diff
@@ -2140,6 +2140,63 @@ find_payment_account(Account *account)
     return NULL;
 }
 
+typedef struct DateBalPair
+{
+    time64 date;
+    gnc_numeric bal;
+} DateBalPair_t;
+
+static int compare_date (void* a, void* b)
+{
+    DateBalPair_t* da = (DateBalPair_t *)a;
+    DateBalPair_t* db = (DateBalPair_t *)b;
+    if (da->date < db->date) return -1;
+    if (da->date > db->date) return 1;
+    return 0;
+}
+
+static void
+AccountSetReconcileDateBalance (Account *acc, time64 date, gnc_numeric balance)
+{
+    GList *list = NULL;
+    gint num_recs, i;
+    DateBalPair_t *datebal;
+
+    if (!acc) return;
+
+    num_recs = xaccAccountGetReconcileNumDates (acc);
+    datebal = g_new0 (DateBalPair_t, 1);
+    datebal->date = date;
+    datebal->bal = balance;
+    list = g_list_prepend (list, datebal);
+
+    for (gint i = 0; i < num_recs; i++)
+    {
+        DateBalPair_t *datebal = g_new0 (DateBalPair_t, 1);
+
+        if (xaccAccountGetReconcileDateBalance (acc, i, &datebal->date, &datebal->bal))
+        {
+            if (datebal->date != date)
+                list = g_list_prepend (list, datebal);
+        }
+        else
+            PWARN ("AccountSetReconcileDateBalance error no rec-info %d exists", i);
+    }
+
+    xaccAccountClearReconcileDates (acc);
+
+    list = g_list_sort (list, (GCompareFunc) compare_date);
+
+    i = 0;
+    for (GList *node = list; node; node = node->next)
+    {
+        datebal = node->data;
+        xaccAccountSetReconcileDateBalance (acc, i, datebal->date, datebal->bal);
+        i++;
+    }
+    xaccAccountSetReconcileNumDates (acc, i);
+    return;
+}
```

Comment:
Poor abstraction. These functions belong in Account.cpp, making the functions in the previous commit static and so hiding the implementation details.